app/assets/javascripts/utilities/showUserAlertModal.js

```diff
@@ -0,0 +1,145 @@
+/**
```

Comment:
More of a side note, but I see why you had difficulties in putting this in Preact land/modern JS code base (ES modules) as it needs to be called from the legacy JS scripts. It would require a `setTimeout` or `setInterval` to determine when the modal component was loaded to be able to use it which is not that ideal.